### ISSUE TITLE
No arm64 image on the 4.3.11 tag (k8s env)

### ISSUE THREAD

**Issue by samip5** [emqx/emqx#6870]
### Subject of the issue
I'm getting exec format error when trying to use 4.3.11 tag on arm64.

### Your environment

- EMQ X version: 4.3.11
- Hardware configuration (e.g. `lscpu`):
```
Architecture:                    aarch64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
Vendor ID:                       ARM
Model:                           3
Model name:                      Cortex-A72
Stepping:                        r0p3
CPU max MHz:                     1500.0000
CPU min MHz:                     600.0000
BogoMIPS:                        108.00
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Vulnerable
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm crc32 cpuid
```
- OS (e.g. `cat /etc/os-release`):
```
AME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Kernel (e.g. `uname -a`): `Linux k8s-master1 5.4.0-1050-raspi #56-Ubuntu SMP PREEMPT Thu Jan 13 13:09:35 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux`
- Others:

$ k logs -n home emqx-2:
standard_init_linux.go:228: exec user process caused: exec format error

$ k describe pod -n home emqx-2:
```
Name:         emqx-2
Namespace:    home
Priority:     0
Node:         k8s-worker6/10.0.105.16
Start Time:   Wed, 26 Jan 2022 12:12:45 +0200
Labels:       app=emqx
              app.kubernetes.io/instance=emqx
              app.kubernetes.io/name=emqx
              controller-revision-hash=emqx-7df9dd5f5f
              statefulset.kubernetes.io/pod-name=emqx-2
              version=4.3.11
Annotations:  checksum/config: 2c02810601ca12884a56244d985b2ff5827aa6ecd973336bbd0da242cacd577f
              cni.projectcalico.org/podIP: 172.16.248.110/32
              cni.projectcalico.org/podIPs: 172.16.248.110/32,fde6:e924:6ef5:f854:cc3c:207:a03f:cae/128
              kubectl.kubernetes.io/restartedAt: 2022-01-26T12:07:09+02:00
Status:       Running
IP:           172.16.248.110
IPs:
  IP:           172.16.248.110
  IP:           fde6:e924:6ef5:f854:cc3c:207:a03f:cae
Controlled By:  StatefulSet/emqx
Containers:
  emqx:
    Container ID:   containerd://18b03240e710144202e004e4ac57960a03c1dbeeac70e4746ebaf74c0ffa241e
    Image:          public.ecr.aws/emqx/emqx:4.3.11
    Image ID:       public.ecr.aws/emqx/emqx@sha256:3d0dc10346d561ee30353b97f1cf4aa0ba72f74c1d91bbf170bbc39297b208aa
    Ports:          1883/TCP, 8883/TCP, 8081/TCP, 8083/TCP, 8084/TCP, 18083/TCP, 4370/TCP
    Host Ports:     0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Wed, 26 Jan 2022 12:24:29 +0200
      Finished:     Wed, 26 Jan 2022 12:24:29 +0200
    Ready:          False
    Restart Count:  7
    Readiness:      http-get http://:8081/status delay=5s timeout=1s period=5s #success=1 #failure=3
    Environment Variables from:
      emqx-env  ConfigMap  Optional: false
    Environment:
      EMQX_NAME:                        emqx
      EMQX_CLUSTER__K8S__APP_NAME:      emqx
      EMQX_CLUSTER__DISCOVERY:          k8s
      EMQX_CLUSTER__K8S__SERVICE_NAME:  emqx-headless
      EMQX_CLUSTER__K8S__NAMESPACE:     home
    Mounts:
      /opt/emqx/data from emqx-data (rw)
      /opt/emqx/data/loaded_modules from emqx-loaded-modules (rw,path="loaded_modules")
      /opt/emqx/data/loaded_plugins from emqx-loaded-plugins (rw,path="loaded_plugins")
      /opt/emqx/etc/acl.conf from emqx-acl (rw,path="acl.conf")
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-mm6nl (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  emqx-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  emqx-data-emqx-2
    ReadOnly:   false
  emqx-loaded-plugins:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      emqx-loaded-plugins
    Optional:  false
  emqx-loaded-modules:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      emqx-loaded-modules
    Optional:  false
  emqx-acl:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      emqx-acl
    Optional:  false
  kube-api-access-mm6nl:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  10m                  default-scheduler  Successfully assigned home/emqx-2 to k8s-worker6
  Normal   Pulling    10m                  kubelet            Pulling image "public.ecr.aws/emqx/emqx:4.3.11"
  Normal   Pulled     10m                  kubelet            Successfully pulled image "public.ecr.aws/emqx/emqx:4.3.11" in 23.969074672s
  Normal   Created    9m28s (x4 over 10m)  kubelet            Created container emqx
  Normal   Started    9m28s (x4 over 10m)  kubelet            Started container emqx
  Normal   Pulled     8m37s (x4 over 10m)  kubelet            Container image "public.ecr.aws/emqx/emqx:4.3.11" already present on machine
  Warning  BackOff    45s (x50 over 10m)   kubelet            Back-off restarting failed container
  ```

Using Helm, <https://github.com/samip5/k8s-cluster/blob/d4be48f57c411a45d5c3854a3c9f6b620fa45deb/cluster/apps/home/emqx/helm-release.yaml>

### Steps to reproduce
Pull the 4.3.11 tag on arm64 node, and have it fail to start due to exec format.

### Expected behaviour
It would have worked without issue.

### Actual behaviour

It doesn't boot.

**Comment by zmstone** [comment]
Hi @samip5 
sorry for the inconvenience, there are maybe something wrong in the docker push CI job towards aws ecr.

I have verified that the image pushed to dockerhub has multi-platform manifest as below.
you can pull the image from dockerhub before we fix it in aws ecr.
```
# docker manifest inspect emqx/emqx:4.3.11
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:f618dfc834fab14911bc37e24dc834cbffb010b8af9e5c6861721ad0429f2a20",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:e8183bc2417503492021c89ec461bf919cb0499dd262dfe78f08003e5f47af3d",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}
```

**Comment by samip5** [comment]
> Hi @samip5 sorry for the inconvenience, there are maybe something wrong in the docker push CI job towards aws ecr.
> 
> I have verified that the image pushed to dockerhub has multi-platform manifest as below. you can pull the image from dockerhub before we fix it in aws ecr.
> 
> ```
> # docker manifest inspect emqx/emqx:4.3.11
> {
>    "schemaVersion": 2,
>    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
>    "manifests": [
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 1786,
>          "digest": "sha256:f618dfc834fab14911bc37e24dc834cbffb010b8af9e5c6861721ad0429f2a20",
>          "platform": {
>             "architecture": "amd64",
>             "os": "linux"
>          }
>       },
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 1786,
>          "digest": "sha256:e8183bc2417503492021c89ec461bf919cb0499dd262dfe78f08003e5f47af3d",
>          "platform": {
>             "architecture": "arm64",
>             "os": "linux"
>          }
>       }
>    ]
> }
> ```

Unfortunately, I'm pretty much always hitting their rate-limit so that's why i try to use docker hub as little as possible so that's not really feasable.

**Comment by samip5** [comment]
And yes, it doesn't seem to have platform info at all:


$ docker manifest inspect public.ecr.aws/emqx/emqx:4.3.11
```json
{
        "schemaVersion": 2,
        "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
        "config": {
                "mediaType": "application/vnd.docker.container.image.v1+json",
                "size": 5373,
                "digest": "sha256:0ef9bc19d70ec1e5d1a32ac2aa57eb2f1057e19c3f7bccb3c37b300b57480560"
        },
        "layers": [
                {
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 2809473,
                        "digest": "sha256:8572bc8fb8a32061648dd183b2c0451c82be1bd053a4ea8fae991436b92faebb"
                },
                {
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 1615,
                        "digest": "sha256:2515d755520291323819dea51dad745ef2b0a7715c89d9ad47cba47af242323d"
                },
                {
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 35052444,
                        "digest": "sha256:5c87fdd888ef3ca7dc3e5947d2b5030dab0c5f4ccf761243eea4ce6bc49b6a51"
                },
                {
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 431,
                        "digest": "sha256:c227b28046716741d0d9c7fe61856a9fd84b56483d0541280e1e6d92a38e63a2"
                },
                {
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 3050212,
                        "digest": "sha256:f3c4e6b224c6b9d1615cefa862058bd154b41c2661cb089e7f7d780dd59287bf"
                },
                {
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 1313,
                        "digest": "sha256:d1d9200332e375da54553eb13d8c828d3a2d36647e1b29ef350ed3e4d5b879c6"
                },
                {
                        "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
                        "size": 35052308,
                        "digest": "sha256:77131c6e337cbff822d6e19eb9078effa7c7b647a2758d55b28582735b24cfba"
                }
        ]
}
```

**Comment by zmstone** [comment]
ref: https://aws.amazon.com/blogs/containers/introducing-multi-architecture-container-images-for-amazon-ecr/

**Comment by zmstone** [comment]
Hi @samip5 It's been fixed for version 4.3.12.

```
$ docker manifest inspect public.ecr.aws/emqx/emqx:4.3.12
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:5245f2a552a0d4d4afd328eda42901e75ce077a0596f062378f13f820961451b",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1786,
         "digest": "sha256:85b861debd27463c9fdd20480e79a3bdb0677b6836c721819f3f9155ad8c6849",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}
```
